internal/human/marshal.go

```diff
@@ -106,16 +105,20 @@ func marshalStruct(value reflect.Value, opt *MarshalOpt) (string, error) {
 		rType := value.Type()
 
 		switch {
+		// If data is nil
+		case isInterfaceNil(value.Interface()):
+			return nil, nil
+
 		// If data has a registered MarshalerFunc call it.
 		case marshalerFuncs[rType] != nil:
 			str, err := marshalerFuncs[rType](value.Interface(), subOpts)
 			return [][]string{{strings.Join(keys, "."), str}}, err
 
+		// If data is a stringers
+		case rType.Implements(reflect.TypeOf((*fmt.Stringer)(nil)).Elem()):
```

Comment:
shouldn't we use a custom `marshallerFunc` ?

Reply:
This is an interface not a type so the marshal func would not work.